src/generate/rust.rs

```diff
@@ -820,32 +820,38 @@ where
     let code_label = Rc::new(CodeLabel::NamedRule(name.to_string()));
     let parse_node_kind = ParseNodeKind::NamedRule(name.to_string());
     let rust_slice_ty = Pat::rust_slice_ty();
-    quote!(impl<'_a, I> #ident<'_a, 'static, I>
-        where I: ::gll::runtime::Input<Slice = #rust_slice_ty>,
-    {
-        pub fn parse_with<R>(
-            input: I,
-            f: impl for<'a, 'i> FnOnce(::gll::runtime::ParseResult<Handle<'a, 'i, I, #ident<'a, 'i, I>>>) -> R,
-        ) -> R {
-            ::gll::runtime::Parser::parse_with(
-                input,
-                #code_label,
-                #parse_node_kind,
-                |result| f(match result {
-                    Ok((ref forest, node)) => Ok(Handle {
-                        node,
-                        forest,
-                        _marker: PhantomData,
-                    }),
-                    Err(ref err) => Err(err.as_ref_partial().map_partial(|&(ref forest, node)| Handle {
+    quote!(
+        impl<'_a, I> #ident<'_a, 'static, I>
+            where I: ::gll::runtime::Input<Slice = #rust_slice_ty>,
+        {
+            pub fn parse(input: I)
+                -> ::gll::runtime::ParseResult<OwnedHandle<I, #ident<'_a, 'static, I>>>
+            {
+                let handle = |forest_and_node| OwnedHandle {
+                    forest_and_node,
+                    _marker: PhantomData,
```

Comment:
Bake away this `PhantomData` into a constructor function?

Reply:
It's *only* supposed to be constructed here.